### PR TITLE
Update the "releasing Airflow" docs

### PR DESCRIPTION
**Pull request by jedcunningham**
I'm finding less and less each cycle 👍

### CODE REVIEW

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Been there, done that.

**Comment by potiuk** [comment]
BTW. At the moment you stop finding them, it's the time to start mentoring a Padawan.

**Review by kaxil — APPROVED** [review]

File: dev/README_RELEASE_AIRFLOW.md [hunk 1]
```diff
@@ -352,7 +352,7 @@ is not supposed to be used by and advertised to the end-users who do not read th
     (both airflow and latest provider packages).
 
     ```shell script
-    git push origin ${VERSION}
+    git push origin tag ${VERSION}
```
Comment (kaxil) [inline dev/README_RELEASE_AIRFLOW.md:355]: The original should work too

Comment (jedcunningham) [reply dev/README_RELEASE_AIRFLOW.md:355]: Indeed, but this is safer.

Comment (uranusjr) [reply dev/README_RELEASE_AIRFLOW.md:355]: TIL this is a thing

Comment (potiuk) [reply dev/README_RELEASE_AIRFLOW.md:355]: Git never ceases to amaze me. Every time I get to do something really complex in git, it really is just a matter of looking hard enough. There has ALWAYS been a one liner (even if a line is long) that does precisely what I wanted to do - no matter how complex the thing was.

**Pull request merged**